SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/OutgoingMessages.h

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+#pragma once
+
+#include <WorkerSDK/improbable/c_worker.h>
+
+enum class EOutgoingMessageType : int32
+{
+	ReserveEntityIdsRequest,
+	CreateEntityRequest,
+	DeleteEntityRequest,
+	ComponentUpdate,
+	CommandRequest,
+	CommandResponse,
+	CommandFailure,
+	LogMessage,
+	ComponentInterest,
+	EntityQueryRequest,
+	Metrics
+};
+
+struct FOutgoingMessage
+{
+	FOutgoingMessage(const EOutgoingMessageType& InType) : Type(InType) {}
+
+	EOutgoingMessageType Type;
+};
+
+struct FReserveEntityIdsRequest : FOutgoingMessage
+{
+	FReserveEntityIdsRequest(const uint32_t& InNumOfEntities)
+		: FOutgoingMessage(EOutgoingMessageType::ReserveEntityIdsRequest)
+		, NumOfEntities(InNumOfEntities)
+	{}
+
+	uint32_t NumOfEntities;
+};
+
+struct FCreateEntityRequest : FOutgoingMessage
+{
+	FCreateEntityRequest(TArray<Worker_ComponentData>&& InComponents, const Worker_EntityId* InEntityId)
+		: FOutgoingMessage(EOutgoingMessageType::CreateEntityRequest)
+		, Components(MoveTemp(InComponents))
+		, EntityId(InEntityId != nullptr ? *InEntityId : TOptional<Worker_EntityId>())
+	{}
+
+	TArray<Worker_ComponentData> Components;
+	TOptional<Worker_EntityId> EntityId;
+};
+
+struct FDeleteEntityRequest : FOutgoingMessage
+{
+	FDeleteEntityRequest(const Worker_EntityId& InEntityId)
+		: FOutgoingMessage(EOutgoingMessageType::DeleteEntityRequest)
+		, EntityId(InEntityId)
+	{}
+
+	Worker_EntityId EntityId;
+};
+
+struct FComponentUpdate : FOutgoingMessage
+{
+	FComponentUpdate(const Worker_EntityId& InEntityId, const Worker_ComponentUpdate& InComponentUpdate)
```

Comment:
Prefer passing primitives by value.
```suggestion
	FComponentUpdate(Worker_EntityId InEntityId, const Worker_ComponentUpdate& InComponentUpdate)
```